Mathlib/Algebra/Lie/LieTheorem.lean

```diff
@@ -0,0 +1,553 @@
+/-
+Copyright (c) 2024 Lucas Whitfield. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lucas Whitfield
+-/
+import Mathlib.Algebra.Lie.Solvable
+import Mathlib.Algebra.Lie.Weights.Linear
+
+/-!
+# Lie's theorem for Solvable Lie algebras.
+
+This file proves Lie's theorem, the statement that Lie modules of solvable Lie algebras over
+algebraically closed fields of characteristic 0 have a common eigenvector for the action of all
+elements of the Lie algebra. This result is named `LieModule.exists_forall_lie_eq_smul_of_Solvable`.
+-/
+
+open LieAlgebra
+
+-- let k be a field of characteristic zero
+variable {k : Type*} [Field k] [CharZero k] [IsAlgClosed k]
```

Comment:
Is it possible to obtain a result without assuming `IsAlgClosed k` if we assume `LieModule.IsTriangularizable k L V`?